### PR TITLE
Introduces caseless ammo, and a caseless rifle

### PR DESCRIPTION
**Pull request by SpadesNeil**
Originally a part of https://github.com/PolarisSS13/Polaris/pull/5837

So let me be very clear about one thing.
## This gun isn't balanced yet.
I need feedback to decide how it will be balanced. Right now, this is just a proof of concept.

The goal here is to make it possible for certain ammunition types to be caseless. I had done this previously in a very old version of Baycode like five years ago, but it was tied to the gun rather than the ammo. In this current version, you could theoretically change _any_ ammo to have a caseless version, and it will work in any gun that chambers it.

The caseless ammo rifle here is intended to be something about as powerful as a pistol caliber, but with a large 30 round magazine, and maybe options for specialized ammo such as stun ammo or EMP ammo in the future. R&D will be able to build this gun.

But like I said, balancing it is an issue, so I need feedback on what materials should be required to make the gun (phoron obviously), and what sort of damage output this gun should have.

### Unrelated but needed changes
- Deleted several unused icons out of gun.dmi

### CODE REVIEW

**Comment by SpadesNeil** [comment]
Also note I tested it thoroughly in the original version of the PR, but have yet to test it independently. Right now, I want to figure out how I am balancing it before I go testing it.

- [ ] What materials should be needed to make the gun?
- [ ] What materials should make the ammo? Phoron, obviously, but what else?
- [ ] How many bullets should be in a magazine? Currently it's 30.
- [ ] Should I add EMP rounds?
- [ ] Should I nerf the stun rounds? Probably.
- [ ] Should I nerf or buff the lethal rounds?

**Comment by Anewbe** [comment]
- Going to specifically request that this not be added anywhere that it can be touched, for the time being. Rather be able to test it through events, badminnery, etc.
- Also, let's **_not_** have this pathed and probably sprited as the G11. Make something new; we have too many IRL guns already, as has been discussed.
- Speaking of sprites, I'm unable to find an item/on-ground sprite for this, which means it will be invisible in the inventory and on the ground.
- 30 rounds is too fucking many. Try 15.
- Scrap the electrode bullet.

**Comment by SpadesNeil** [comment]
Did I not add an item sprite to gun.dmi??

**Comment by SpadesNeil** [comment]
I did. Apparently gun.dmi is literally at its limit of how many sprites can go into it.

WELP. Guess it's time to start nuking unused shit.

**Comment by SpadesNeil** [comment]
ToDo: Move energy/projectile guns into their own files.

**Comment by Woodratt** [comment]
Would agree to it being admin spawn only for now until it can be messed around with and see what it does proper like. In the future it could be added to research, but that'd be a thing for the future I'd say.

As it stands, leave the sprites as is I suppose, though from the sounds of things looks like the gun dmi needs to be broken up. 

Leave the electrode bullet and 30 round magazine I'd say. If the damage is comparable to a 9mm pistol, then its no way near as powerful as say, a STS right now, or even the basic SMG. Both of those have burst fire options to increase DPS which this does not. 

Electrode round doesn't really matter if it exists or not if its admin spawn only. We can decide later if players can get their hands on it.

**Comment by Anewbe** [comment]
I will repeat: I don't care what the Real-World Equivalent is, I carry that this HAS a Rea-World Equivalent, AND that it's patched as such. I don't want to have to deal with more "But [gun] does [thing] IRL" conversations.

**Comment by SpadesNeil** [comment]
Okay, well, it's not pathed as a G11 anymore.

File: code/modules/projectiles/ammunition/magazines.dm [hunk 1]
```diff
@@ -188,6 +188,21 @@
 	name = "speedloader (.45 AP)"
 	ammo_type = /obj/item/ammo_casing/a45/ap
 
+///////// 5mm Caseless /////////
+
+/obj/item/ammo_magazine/m5mmcaseless
+	name = "prototype rifle magazine (5mm caseless)"
+	ammo_type = /obj/item/ammo_casing/a5mmcaseless
+	icon_state = "G11mag"
```
Comment (SpadesNeil) [inline code/modules/projectiles/ammunition/magazines.dm]: Wait, I lied, I didn't do the mags.

**Comment by SpadesNeil** [comment]
As an aside, there's a _lot_ of redundant and unused stuff in ammo.dmi

Remind me to sort all those out in a later PR.

**Comment by SpadesNeil** [comment]
I'm _trying_ to make some more visually distinct examples of this rifle, but I am a really shitty sprite artist.

I don't want to throw out the other _twelve_ in-hand sprites, while at the same time, I refuse to do what ArmA 3 did and present a lazily done generic assault rifle that just doesn't spit out shells when fired.

This is what I have so far...
caseless-alt

**Comment by SpadesNeil** [comment]
Gonna wait until https://github.com/PolarisSS13/Polaris/pull/5881 is merged, then fix the conflicts manually.

**Comment by SpadesNeil** [comment]
I forgot what a bitch Github could be with .dmi conflicts.

**Comment by SpadesNeil** [comment]
For fuck's sake. Starting again.

**Comment by SpadesNeil** [comment]
This, kids, is why I do a bunch of different things all in one PR.

If not for the bullshit that Github gives me whenever I dare to _touch_ a .dmi file, I would not have this problem.

**Comment by SpadesNeil** [comment]
Conflict resolved.

muffled autistic screeching

**Pull request merged**